etna/models/base.py

```diff
@@ -151,43 +145,68 @@ def get_model(self) -> Union[Any, Dict[str, Any]]:
         pass
 
 
+class PredictionIntervalAbstractModel(ABC):
+    """Interface that is used to mark classes that support prediction intervals."""
+
+    pass
+
+
+class ContextRequiredAbstractModel(ABC):
+    """Interface that is used to mark classes that need context for prediction."""
+
+    @property
+    @abstractmethod
+    def context_size(self) -> int:
+        """Context size of the model. Determines how many history points do we ask to pass to the model."""
+        pass
+
+
 class ForecastAbstractModel(ABC):
     """Interface for model with forecast method."""
 
-    @abstractmethod
-    def forecast(self, ts: TSDataset) -> TSDataset:
-        """Make predictions.
+    def _extract_prediction_interval_params(self, **kwargs) -> Dict[str, Any]:
+        extracted_params = {}
+        class_name = self.__class__.__name__
 
-        Parameters
-        ----------
-        ts:
-            Dataset with features
+        if isinstance(self, PredictionIntervalAbstractModel):
+            prediction_interval = kwargs.get("prediction_interval", False)
+            extracted_params["prediction_interval"] = prediction_interval
 
-        Returns
-        -------
-        :
-            Dataset with predictions
-        """
-        pass
+            quantiles = kwargs.get("quantiles", (0.025, 0.975))
+            extracted_params["quantiles"] = quantiles
+        else:
+            if "prediction_interval" in kwargs or "quantiles" in kwargs:
+                raise NotImplementedError(f"Model {class_name} doesn't support prediction intervals!")
+
+        return extracted_params
 
+    def _extract_prediction_size_params(self, **kwargs):
+        extracted_params = {}
+        class_name = self.__class__.__name__
 
-class PredictIntervalAbstractModel(ABC):
-    """Interface for model with forecast method that creates prediction interval."""
+        if isinstance(self, ContextRequiredAbstractModel):
+            prediction_size = kwargs.get("prediction_size")
+            if prediction_size is None:
+                raise ValueError(f"Parameter prediction_size is required for {class_name} model!")
+
+            if not isinstance(prediction_size, int) or prediction_size <= 0:
+                raise ValueError(f"Parameter prediction_size should be positive integer!")
+
+            extracted_params = prediction_size
+        else:
+            if "prediction_size" in kwargs:
+                raise NotImplementedError(f"Model {class_name} doesn't support prediction_size parameter!")
+
+        return extracted_params
 
     @abstractmethod
-    def forecast(
-        self, ts: TSDataset, prediction_interval: bool = False, quantiles: Sequence[float] = (0.025, 0.975)
-    ) -> TSDataset:
+    def forecast(self, ts: TSDataset, **kwargs) -> TSDataset:
```

Comment:
I like this solution more, however we do't state the parameters of `forecast` method for different types of the models. May be we need to say sometzid about it in the docstrings for `ContextRequiredAbstractModel ` and `PredictionIntervalAbstractModel `